lib/ansible/modules/cloud/amazon/lambda_policy.py

```diff
@@ -0,0 +1,529 @@
+    - aws
+
+'''
+
+EXAMPLES = '''
+---
+- hosts: localhost
+  gather_facts: no
+  vars:
+    state: present
+  tasks:
+  - name: Lambda S3 event notification
+    lambda_policy:
+      state: "{{ state | default('present') }}"
+      function_name: functionName
+      alias: Dev
+      statement_id: lambda-s3-myBucket-create-data-log
+      action: lambda:InvokeFunction
+      principal: s3.amazonaws.com
+      source_arn: arn:aws:s3:eu-central-1:123456789012:bucketName
+      source_account: 123456789012
+
+  - name: show results
+    debug: var=lambda_policy_action
+
+'''
+
+RETURN = '''
+---
+lambda_policy_action:
+    description: describes what action was taken
+    returned: success
+    type: string
+'''
+
+import json
+import re
+import traceback
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.ec2 import HAS_BOTO3, get_aws_connection_info, boto3_conn, ec2_argument_spec, camel_dict_to_snake_dict
+
+
+# ---------------------------------------------------------------------------------------------------
+#
+#   Helper Functions & classes
+#
+# ---------------------------------------------------------------------------------------------------
+
+class AWSConnectionException(Exception):
+    pass
+
+
+class AWSConnection:
+    """
+    Create the connection object and client objects as required.
+    """
+
+    def __init__(self, ansible_obj, resources, boto3=True):
+
+        try:
+            self.region, self.endpoint, aws_connect_kwargs = get_aws_connection_info(
+                ansible_obj, boto3=boto3)
+            if not self.region:
+                raise AWSConnectionException('region must be specified')
+
+            self.resource_client = dict()
+            if not resources:
+                resources = ['lambda']
+
+            resources.append('iam')
+
+            for resource in resources:
+                aws_connect_kwargs.update(dict(region=self.region,
+                                               endpoint=self.endpoint,
+                                               conn_type='client',
+                                               resource=resource
+                                               ))
+                self.resource_client[resource] = boto3_conn(ansible_obj, **aws_connect_kwargs)
+
+        except Exception as e:
+            fail_json_aws(ansible_obj, e, msg="setting up AWS connection")
+
+    def client(self, resource='lambda'):
+        return self.resource_client[resource]
+
+
+def pc(key):
+    """
+    Changes python key into Pascale case equivalent. For example, 'this_function_name' becomes 'ThisFunctionName'.
+
+    :param key:
+    :return:
+    """
+
+    return "".join([token.capitalize() for token in key.split('_')])
+
+
+def policy_equal(module, current_statement):
+
+    equal = True
+
+    for param in ('action', 'principal', 'source_arn', 'source_account', 'event_source_token'):
+        if module.params.get(param) != current_statement.get(param):
```

Comment:
So the issue I was thinking might come up was that the API may "fix" lists of one and only return the list content (as S3 does). So running a playbook with a list of one item always displayed changed. I looked at the boto3 docs and since these options must be strings it's a non-issue.

Testing this out though, I notice that if an invalid option is provided (such as specifying a list of two actions for a statement) or a typo results in a failure (good, as expected) but also deletes the existing policy (that seems weird). I don't know if that's easily fixable in a way other than validating the options before making API calls...
```
    "error": {
        "code": "ValidationException",
        "message": "1 validation error detected: Value '['lambda:CreateFunction']' at 'action' failed to satisfy constraint: Member must satisfy regular expression pattern: (lambda:[*]|lambda:[a-zA-Z]+|[*])"
    },
```